lib/Cluster.js

```diff
@@ -991,6 +1058,42 @@ class Cluster extends EventEmitter {
     }
   }
 
+  /**
+   * Given an array of attribute ids it checks if all of the attributes are manufacturer
+   * specific attributes. If that is the case it will return the manufacturerId so that the
+   * command can set the manufacturerSpecific flag.
+   * @param {number[]} attributeIds
+   * @returns {null|number}
+   * @private
+   */
+  _checkForManufacturerSpecificAttributes(attributeIds) {
+    // Convert to set
+    const attrIdsSet = new Set(attributeIds);
+
+    // Filter attributeIds for manufacturer specific attributes
+    const manufacturerIds = Object.entries(this.constructor.attributes)
+      .reduce((acc, [key, value]) => {
+        if (attrIdsSet.has(value.id) && typeof value.manufacturerId === 'number') acc.push(value.manufacturerId);
+        return acc;
+      }, []);
```

Comment:
I generally prefer not to use reduce since it is harder to read/understand (for me at least).

```js
const manufacturerIds = [];
for (const attribute of Object.values(this.constructor.attributes)) {
  if (attrIdsSet.has(attribute.id) && typeof attribute.manufacturerId === 'number') {
    manufacturerIds.push(attribute.manufacturerId);
  }
}

Reply:
Your suggestion is more readable indeed, will fix that.